sdk/hybridnetwork/Azure.ResourceManager.HybridNetwork/src/Generated/Models/SecretDeploymentResourceReference.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.HybridNetwork.Models
+{
+    /// <summary> Secret deployment resource id reference. </summary>
+    public partial class SecretDeploymentResourceReference : DeploymentResourceIdReference
+    {
+        /// <summary> Initializes a new instance of SecretDeploymentResourceReference. </summary>
+        public SecretDeploymentResourceReference()
+        {
+            IdType = IdType.Secret;
+        }
+
+        /// <summary> Initializes a new instance of SecretDeploymentResourceReference. </summary>
+        /// <param name="idType"> The resource reference arm id type. </param>
+        /// <param name="id"> Resource ID. </param>
+        internal SecretDeploymentResourceReference(IdType idType, string id) : base(idType)
+        {
+            Id = id;
+            IdType = idType;
+        }
+
+        /// <summary> Resource ID. </summary>
+        public string Id { get; set; }
```

Comment:
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/maps/Azure.ResourceManager.Maps/src/autorest.md?plain=1#L25).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/search/Azure.Search.Documents/src/autorest.md?plain=1#L106-L115).